ofproto/ofproto-dpif-xlate.c

```diff
@@ -3062,6 +3127,19 @@ xlate_normal(struct xlate_ctx *ctx)
         update_learning_table(ctx, in_xbundle, flow->dl_src, vlan,
                               is_grat_arp);
     }
+
+#if defined(P4OVS)
+    //MAC is learnt, program P4 forwarding table
+    struct xport *ovs_port = get_ofp_port(in_xbundle->xbridge,
+                                          flow->in_port.ofp_port);
+    struct mac_learning_info fdb_info;
+    memset(&fdb_info, 0, sizeof(fdb_info));
+
+    if (!get_fdb_data(ovs_port, flow->dl_src, &fdb_info)) {
```

Comment:
Should this function continue if `get_fdb_data` returns an error?

Reply:
Added a log in case get_fdb_data returns error and also, yes function should continue. We don't want to limit OVS function in case of failure with P4 related code